sycl/source/handler.cpp

```diff
@@ -312,6 +312,25 @@ fill_copy_args(detail::handler_impl *impl,
 
 } // namespace detail
 
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+
+handler::handler(const std::shared_ptr<detail::queue_impl> &Queue,
+                 bool CallerNeedsEvent)
+    : MImplOwner(std::make_shared<detail::handler_impl>(Queue.get(),
+                                                        CallerNeedsEvent)),
+      impl(MImplOwner.get()), MQueue(Queue) {}
+
+handler::handler(detail::handler_impl *HandlerImpl,
+                 const std::shared_ptr<detail::queue_impl> &Queue)
+    : impl(HandlerImpl), MQueue(Queue) {}
+
+static std::shared_ptr<detail::queue_impl> DummyQueue;
```

Comment:
This static variable is a bit odd as it's global, and could probably use a comment. e.g. something like "need a const reference to queue_impl shared pointer to initalize MQueue with"

I'm also wondering if it could be defined inside `graph_impl::add()` used to call `handler::handler(detail::handler_impl *HandlerImpl, const std::shared_ptr<detail::queue_impl> &Queue)` directly. That might remove the need to define the `handler::handler(detail::handler_impl *HandlerImpl)` constructor and avoid the reader needing to mentally connect this dummy variable in handler.cpp with the call site in graph_impl.cpp

Reply:
Thank you, let's decrease number on constructors.